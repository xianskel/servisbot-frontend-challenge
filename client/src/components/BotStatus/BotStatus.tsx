import { Tag } from "primereact/tag";
import { useMediaQuery } from "usehooks-ts";
import { Status } from "../../types";

type Severity = "success" | "danger" | "warning";

const BotStatus = ({ status }: { status: Status }): JSX.Element => {
  const isMobileScreen = useMediaQuery("(max-width: 768px)");
  const statusMap: Record<
    Status,
    { severity: Severity; icon: string; color: string }
  > = {
    DISABLED: {
      severity: "danger",
      icon: "pi pi-exclamation-triangle",
      color: "var(--red-500)",
    },
    ENABLED: {
      severity: "success",
      icon: "pi pi-check-circle",
      color: "var(--green-500)",
    },
    PAUSED: {
      severity: "warning",
      icon: "pi pi-pause-circle",
      color: "var(--orange-500)",
    },
  };

  const mapping = statusMap[status];

  return isMobileScreen ? (
    <i className={mapping?.icon} style={{ color: mapping?.color }}></i>
  ) : (
    <Tag icon={mapping?.icon} severity={mapping?.severity} value={status}></Tag>
  );
};

export default BotStatus;
