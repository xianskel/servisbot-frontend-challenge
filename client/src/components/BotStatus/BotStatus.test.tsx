import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";
import BotStatus from "./BotStatus";

describe("BotStats", () => {
  test("renders correct for status DISABLED", () => {
    const { asFragment } = render(<BotStatus status="DISABLED"></BotStatus>);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders correct for status ENABLED", () => {
    const { asFragment } = render(<BotStatus status="ENABLED"></BotStatus>);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders correct for status PAUSED", () => {
    const { asFragment } = render(<BotStatus status="PAUSED"></BotStatus>);
    expect(asFragment()).toMatchSnapshot();
  });
});
