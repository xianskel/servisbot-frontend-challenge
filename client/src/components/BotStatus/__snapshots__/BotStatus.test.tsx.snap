// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BotStats > renders correct for status DISABLED 1`] = `
<DocumentFragment>
  <span
    class="p-tag p-component p-tag-danger"
    data-pc-name="tag"
    data-pc-section="root"
  >
    <span
      class="p-tag-icon pi pi-exclamation-triangle"
      data-pc-section="icon"
    />
    <span
      class="p-tag-value"
      data-pc-section="value"
    >
      DISABLED
    </span>
    <span />
  </span>
</DocumentFragment>
`;

exports[`BotStats > renders correct for status ENABLED 1`] = `
<DocumentFragment>
  <span
    class="p-tag p-component p-tag-success"
    data-pc-name="tag"
    data-pc-section="root"
  >
    <span
      class="p-tag-icon pi pi-check-circle"
      data-pc-section="icon"
    />
    <span
      class="p-tag-value"
      data-pc-section="value"
    >
      ENABLED
    </span>
    <span />
  </span>
</DocumentFragment>
`;

exports[`BotStats > renders correct for status PAUSED 1`] = `
<DocumentFragment>
  <span
    class="p-tag p-component p-tag-warning"
    data-pc-name="tag"
    data-pc-section="root"
  >
    <span
      class="p-tag-icon pi pi-pause-circle"
      data-pc-section="icon"
    />
    <span
      class="p-tag-value"
      data-pc-section="value"
    >
      PAUSED
    </span>
    <span />
  </span>
</DocumentFragment>
`;
